lib/infrastructure/dex_token.repository.dart

```diff
@@ -49,100 +53,96 @@ class DexTokenRepositoryImpl with ModelParser implements DexTokenRepository {
 
+      final token1Address = value.properties['token1_address'];
+      final token2Address = value.properties['token2_address'];
+
       final _token = value.copyWith(address: key);
 
       var balanceAmount = 0.0;
       for (final tokenBalance in balance.token) {
         if (tokenBalance.address!.toUpperCase() ==
             _token.address!.toUpperCase()) {
-          balanceAmount = fromBigInt(tokenBalance.amount).toDouble();
+          balanceAmount = archethic.fromBigInt(tokenBalance.amount).toDouble();
           break;
         }
       }
 
       var dexToken = tokenSDKToModel(_token, balanceAmount);
 
-      final tokenSymbolSearch = <String>[];
-      if (value.properties.isNotEmpty &&
-          value.properties['token1_address'] != null &&
-          value.properties['token2_address'] != null) {
-        if (value.properties['token1_address'] != 'UCO') {
-          tokenSymbolSearch.add(value.properties['token1_address']);
-        }
-        if (value.properties['token2_address'] != 'UCO') {
-          tokenSymbolSearch.add(value.properties['token2_address']);
-        }
-        final tokensSymbolMap = await aedappfm.sl.get<ApiService>().getToken(
-              tokenSymbolSearch,
-              request: 'name, symbol',
-            );
-
-        dexToken = dexToken.copyWith(
-          isLpToken: true,
-          lpTokenPair: DexPair(
-            token1: value.properties['token1_address'] != 'UCO'
-                ? DexToken(
-                    address: value.properties['token1_address'],
-                    name: tokensSymbolMap[value.properties['token1_address']] !=
-                            null
-                        ? tokensSymbolMap[value.properties['token1_address']]!
-                            .name!
-                        : '',
-                    symbol: tokensSymbolMap[
-                                value.properties['token1_address']] !=
-                            null
-                        ? tokensSymbolMap[value.properties['token1_address']]!
-                            .symbol!
-                        : '',
-                  )
-                : ucoToken,
-            token2: value.properties['token2_address'] != 'UCO'
-                ? DexToken(
-                    address: value.properties['token2_address'],
-                    name: tokensSymbolMap[value.properties['token2_address']] !=
-                            null
-                        ? tokensSymbolMap[value.properties['token2_address']]!
-                            .name!
-                        : '',
-                    symbol: tokensSymbolMap[
-                                value.properties['token2_address']] !=
-                            null
-                        ? tokensSymbolMap[value.properties['token2_address']]!
-                            .symbol!
-                        : '',
-                  )
-                : ucoToken,
-          ),
-        );
-      }
+      dexToken = dexToken.copyWith(
```

Comment:
Only if it's LP Token. But in tokenMap we can have token. Perhaps add condition if token1Address != null && token2Address != null